trade-finance-manager-api/src/v1/controllers/tasks.controller.js

```diff
@@ -77,26 +77,38 @@ const updateTasksCanEdit = async (allTaskGroups, groupId, taskIdToUpdate, deal,
 
     group = {
       ...group,
-      groupTasks: group.groupTasks.map((t) => {
+      groupTasks: group.groupTasks.map((t, index, array) => {
         const task = t;
-
         if (task.status === CONSTANTS.TASKS.STATUS.COMPLETED) {
           task.canEdit = false;
-        } else if (previousTaskIsComplete(allTaskGroups, group, task.id)) {
+        } else if (taskStatusValidity(allTaskGroups, group, task.id, task.group)) {
           if (task.id === taskIdToUpdate
             && task.groupId === groupId) {
             if (task.status === CONSTANTS.TASKS.STATUS.COMPLETED) {
               task.canEdit = false;
             }
           } else {
+            // if Complete an adverse history check task is complete
+            // opens up all underwriting tasks to be editable and set to 'TO DO'
+            if (task.groupId === CONSTANTS.TASKS.MIA.GROUP_3.id
+               && task.title === CONSTANTS.TASKS.MIA_GROUP_3_TASKS.CHECK_EXPOSURE
+               && task.status === CONSTANTS.TASKS.STATUS.CANNOT_START) {
+              for (let i = index + 1; i <= index + 2; i += 1) {
+                const nextTask = array[i];
+                nextTask.canEdit = true;
+                nextTask.status = CONSTANTS.TASKS.STATUS.TO_DO;
+                // Send task notification emails
+                sendUpdatedEmailRequests.push(sendUpdatedTaskEmail(nextTask, deal, urlOrigin));
+              }
```

Comment:
I think this could be cleaner with no need for a for loop and index based, leave with me 😄